src/topology/subset_properties.lean

```diff
@@ -1244,6 +1272,106 @@ begin
+  { split,
+    { exact mt or.inl h2 },
+    rw [set.inter_comm, set.ne_empty_iff_nonempty],
+    apply set.inter_compl_nonempty_iff.2,
+    push_neg at h2,
+    exact h2.2 },
+  rw [set.inter_comm, set.ne_empty_iff_nonempty] at h3,
+  conv at h3 in (sᶜ ∩ t ≠ ∅) {rw [set.inter_comm, set.ne_empty_iff_nonempty]},
+  apply set.ne_empty_iff_nonempty.2 (h s sᶜ h1.1 (is_open_compl_iff.2 h1.2) _ h3.1 h3.2),
+  rw [set.inter_compl_self, set.inter_empty],
+  rw [set.union_compl_self],
+  exact set.subset_univ t,
+end
+
+/-- A set `s` is preconnected if and only if
+for every cover by two closed sets that are disjoint on `s`,
+it is contained in one of the two covering sets -/
+theorem is_preconnected_iff_subset_of_disjoint_closed {α : Type*} {s : set α} [topological_space α] :
+  is_preconnected s ↔
+  ∀ (u v : set α) (hu : is_closed u) (hv : is_closed v) (hs : s ⊆ u ∪ v) (huv : s ∩ (u ∩ v) = ∅),
+  s ⊆ u ∨ s ⊆ v :=
+begin
+  split; intro h,
+  { intros u v hu hv hs huv,
+    rw is_preconnected_closed_iff at h,
+    specialize h u v hu hv hs,
+    contrapose! huv,
+    rw set.ne_empty_iff_nonempty,
+    simp [set.not_subset] at huv,
+    rcases huv with ⟨⟨x, hxs, hxu⟩, ⟨y, hys, hyv⟩⟩,
+    have hxv : x ∈ v := or_iff_not_imp_left.mp (hs hxs) hxu,
+    have hyu : y ∈ u := or_iff_not_imp_right.mp (hs hys) hyv,
+    exact h ⟨y, hys, hyu⟩ ⟨x, hxs, hxv⟩ },
+  { rw is_preconnected_closed_iff,
+    intros u v hu hv hs hsu hsv,
+    rw ← set.ne_empty_iff_nonempty,
+    intro H,
+    specialize h u v hu hv hs H,
+    contrapose H,
+    apply set.ne_empty_iff_nonempty.mpr,
+    cases h,
+    { rcases hsv with ⟨x, hxs, hxv⟩, exact ⟨x, hxs, ⟨h hxs, hxv⟩⟩ },
+    { rcases hsu with ⟨x, hxs, hxu⟩, exact ⟨x, hxs, ⟨hxu, h hxs⟩⟩ } }
+end
+
+/-- A closed set `s` is preconnected if and only if
+for every cover by two closed sets that are disjoint,
+it is contained in one of the two covering sets -/
+theorem is_preconnected_iff_subset_of_fully_disjoint_closed {s : set α} (hs : is_closed s) :
+  is_preconnected s ↔
+  ∀ (u v : set α) (hu : is_closed u) (hv : is_closed v) (hss : s ⊆ u ∪ v) (huv : u ∩ v = ∅),
+  s ⊆ u ∨ s ⊆ v :=
+begin
+  split,
+  { intros,
+    apply is_preconnected_iff_subset_of_disjoint_closed.1 ᾰ u v hu hv hss,
+    rw huv,
+    exact set.inter_empty s },
+  intro H,
+  rw is_preconnected_iff_subset_of_disjoint_closed,
+  intros u v hu hv hss huv,
+  have H1 := H (u ∩ s) (v ∩ s),
+  rw [(@set.subset_inter_iff _ u s s), (@set.subset_inter_iff _ v s s)] at H1,
+  simp only [set.subset.refl, and_true] at H1,
+  apply H1 (is_closed_inter hu hs) (is_closed_inter hv hs),
+  { rw ←set.inter_distrib_right,
+    apply set.subset_inter_iff.2,
+    split,
+      exact hss,
+    exact set.subset.refl s },
+  { conv in (v ∩ s) {rw set.inter_comm},
+    rw set.inter_assoc,
+    conv in (s ∩ (s ∩ v)) {rw [←set.inter_assoc, set.inter_self s]},
+    rw [set.inter_comm, set.inter_assoc],
+    conv in (v ∩ u) {rw set.inter_comm},
+    exact huv }
```

Comment:
On the first line, you can notice that `set.inter_comm` takes explicit arguments. So, instead of doing a `conv` to specify which rewriting you want to do, it is enough to do `rw [set.inter_comm v s]`. Then this rewrite can be combined with the next one, and also the next one, and so on. With this trick, the proof becomes a single rewrite!
```suggestion
  { rw [set.inter_comm v s, set.inter_assoc, ←set.inter_assoc s, set.inter_self s,
        set.inter_comm, set.inter_assoc, set.inter_comm v u, huv] }
```